dhall-docs/src/Dhall/Docs.hs

```diff
@@ -106,20 +119,32 @@ getAllDhallFilesAndHeaders
     :: Path Abs Dir -- ^ Base directory to do the search
     -> IO [(Path Abs File, Header)]
 getAllDhallFilesAndHeaders baseDir = do
-    files <- snd <$> Path.IO.listDirRecur baseDir
+    files <- filter hasDhallExtension . snd <$> Path.IO.listDirRecur baseDir
     Data.Maybe.catMaybes <$> mapM readDhall files
   where
+    hasDhallExtension :: Path Abs File -> Bool
+    hasDhallExtension absFile = case Path.splitExtension absFile of
+        Nothing -> False
+        Just (_, ext) -> ext == ".dhall"
+
     readDhall :: Path Abs File -> IO (Maybe (Path Abs File, Header))
     readDhall absFile = do
         let filePath = Path.fromAbsFile absFile
-        fileContents <- Data.ByteString.readFile filePath
-        return $ case Text.Encoding.decodeUtf8' fileContents of
-            Left _ -> Nothing
-            Right contents ->
-                case exprAndHeaderFromText filePath contents of
-                    Right (header, _) -> return (absFile, header)
-                    _ -> Nothing
-
+        contents <- Text.IO.readFile filePath
+        case exprAndHeaderFromText filePath contents of
+            Right (header, _) -> return $ Just (absFile, removeComments header)
+            Left ParseError{..} -> do
+                putStrLn $ showParseError unwrap
+                return Nothing
+
+    removeComments :: Header -> Header
```

Comment:
Maybe call this `stripComment` or `stripCommentSyntax`?!

Reply:
The type seems wrong – the output isn't a valid `Header` anymore! Maybe output `Text` or a dedicated newtype?

Reply:
I tend to forget (rookie haskell errors, I think) that ADT represents stuff in real world and as well they should be valid. You're right, output is not a `Header` anymore.

I think I'll add a dedicated type for this purpose

Reply:
No worries! :)

Reply:
I ended up returning just `Text`, but I'm sure that in the future probably I'll have a dedicated data type to represent _extracted documentation_